Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm

```diff
@@ -2439,6 +2439,26 @@ HTTPServer server({
     checkFrameTreesInProcesses(opened.webView.get(), { { "https://webkit.org"_s } });
 }
 
+TEST(SiteIsolation, NavigateOpenerToProvisionalNavigationFailure)
+{
+    HTTPServer server({
+        { "/example"_s, { "<script>w = window.open('https://webkit.org/webkit')</script>"_s } },
+        { "/webkit"_s, { "hi"_s } },
+        { "/terminate"_s, { HTTPResponse::TerminateConnection::Yes } }
+    }, HTTPServer::Protocol::HttpsProxy);
+
+    auto [opener, opened] = openerAndOpenedViews(server);
+    checkFrameTreesInProcesses(opener.webView.get(), { { "https://example.com"_s }, { RemoteFrame } });
+    checkFrameTreesInProcesses(opened.webView.get(), { { RemoteFrame }, { "https://webkit.org"_s } });
+
+    [opened.webView evaluateJavaScript:@"opener.location = '/terminate'" completionHandler:nil];
```

Comment:
Maybe it's worth having another test to navigate the opener to a different domain?

Reply:
oh, there are a lot more tests needed: redirects, iframes, etc.  That's why this PR is named "Begin".  There's still a lot of work to do